src/main/java/com/autotune/experimentManager/data/TrialInterfaceImpl.java

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat, IBM Corporation and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.autotune.experimentManager.data;
+
+import com.autotune.common.experiments.ExperimentTrial;
+import com.autotune.common.experiments.TrialDetails;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Experiment's Local storage Data access implementation .
+ */
+public class TrialInterfaceImpl implements TrialInterface {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TrialInterfaceImpl.class);
+    private final ExperimentDetailsMap<String, ExperimentTrial> existingExperimentTrialMap;
```

Comment:
`existingExperimentTrialMap` This needs a better name. How about either `experimentTrialMap` or `EMExperimentTrialMap`